### ISSUE TITLE
Support tsconfig.extends as `array`

### ISSUE THREAD

**Issue by hubertkuoch** [typescript-eslint/typescript-eslint#6219]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```typescript
npm eslint --config .eslintrc.cjs .
// or 
yarn eslint --config .eslintrc.cjs .
// or
pnpm eslint --config .eslintrc.cjs .
```
```


### ESLint Config

```javascript
module.exports = {
  parser: '@typescript-eslint/parser',
  extends: ['eslint:recommended', 'plugin:@typescript-eslint/recommended'],
}
```

### tsconfig

```jsonc
{
  "extends": ["tsconfig.json"]
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Since [tsconfig.extends now accepts array](https://github.com/microsoft/TypeScript/issues/29118), ESLint should also be able to parse it and run normally.

### Actual Result

Parsing error: Compiler option 'extends' requires a value of type string

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.42.0` |
| `@typescript-eslint/parser`            | `5.42.0` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `TypeScript`                           | `5.0.0-dev.20221214` |
| `ESLint`                               | `8.26.0` |
| `node`                                 | `18.12.1` |

**Comment by bradzacher** [comment]
We do not support "dev" versions of TS. We will assess this as part of the 5.0 release.

https://typescript-eslint.io/maintenance/versioning#typescript
